### PR TITLE
refactor(@angular-devkit/build-angular): move dev-server webpack config in a separate file

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/dev-server/index.ts [hunk 1]
```diff
@@ -89,12 +90,12 @@ export function serveWebpackBrowser(
   async function setup(): Promise<{
     browserOptions: json.JsonObject & BrowserBuilderSchema;
     webpackConfig: webpack.Configuration;
-    webpackDevServerConfig: WebpackDevServer.Configuration;
     projectRoot: string;
     locale: string | undefined;
   }> {
     // Get the browser configuration from the target name.
     const rawBrowserOptions = await context.getTargetOptions(browserTarget);
+    options.port = await checkPort(options.port || 4200, options.host || 'localhost');
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/dev-server/index.ts:98]: `0` is a valid value for port but is also falsy so `||` can't be used.

File: packages/angular_devkit/build_angular/src/dev-server/ssl_spec.ts [hunk 2]
```diff
@@ -102,8 +102,8 @@ describe('Dev Server Builder ssl', () => {
 
     const overrides = {
       ssl: true,
-      sslKey: '../ssl/server.key',
-      sslCert: '../ssl/server.crt',
+      sslKey: 'ssl/server.key',
+      sslCert: 'ssl/server.crt',
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/dev-server/ssl_spec.ts:106]: Did this test not work before?

**Review by clydin — CHANGES_REQUESTED** [review]

File: packages/angular_devkit/build_angular/src/dev-server/ssl_spec.ts (hunk 2 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/dev-server/ssl_spec.ts:106]: Yeah, wrong paths

File: packages/angular_devkit/build_angular/src/dev-server/index.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/dev-server/index.ts:98]: That is correct, though that is the current behaviour. I’ll update to use `??`

File: packages/angular_devkit/build_angular/src/dev-server/ssl_spec.ts (hunk 2 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/dev-server/ssl_spec.ts:106]: Perviously, wrong paths errors where swallowed.

**Review by clydin — APPROVED** [review]

**Pull request merged**